### PR TITLE
Append package top-level key options

### LINKED ISSUE #50: Custom build does not append package top-level key options
# Bug Report

### What is expected to happen?

Custom build allows the user to define package and its set of options instructing electron-builder on how it should build that package.

For example:
For the DMG package, you can provide a set of options on how to build the DMG package. Such as background, format and etc. Please, find out more in the [Electron Builder documentation](https://www.electron.build/configuration/dmg).

### What does actually happen?

All additional package options are ignored. 

## Information

Create a custom `build.json` file in the root directory.
```
{
  "electron": {
    "macos": {
      "package": [
        {
         "dmg": { "backgroundColor": "#000000" }
        }
      ],
      "arch": ["x64"]
    }
  }
}
```

### Command or Code

Run the following command:
```
cordova build electron
```

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

### PR DESCRIPTION
**Pull request by GedasGa**
### Platforms affected

electron

### Motivation and Context

Fixes #50 

### Description

Allow to create the top-level mac, linux, windows key contains a set of options instructing electron-builder on how it should build macOS, Linux ar Windows targets. 

### Testing

```
npm t 
```

Also tested the following scenario.

Created build.json:
```
{
  "electron": {
    "linux": {
      "package": [
        "tar.bz2",
        { "appImage": {
          "systemIntegration": "doNotAsk"
         }
        }
      ],
      "arch": ["x64"]
    },
    "windows": {
      "package": ["portable"],
      "arch": ["x64"]
    }
  }
}
```

Ran: 
```
cordova build electron
```

Which resulted in the following `builder-effective-config.yaml` file:
```
...
linux:
  target:
    - target: tar.bz2
      arch:
        - x64
    - target: appImage
      arch:
        - x64
  icon: installer.png
appImage:
  systemIntegration: doNotAsk
win:
  target:
    - target: portable
      arch:
        - x64
  icon: installer.png
```

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Comment by macskay** [comment]
This PR would be really nice to have. At the moment the name of the executable in linux builds is the appId because the option `executableName` as defined in electron documentation can not be set without getting overwritten.

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 1]
```diff
@@ -699,6 +699,152 @@ describe('Testing build.js:', () => {
             expect(electronBuilder.userBuildSettings.config.win).toEqual(expectedWin);
         });
 
+        it('should not append package top-level key options if object is empty.', () => {
+            // mock platformConfig, buildConfig and buildOptions Objects
+            const platformConfig = {
+                mac: { package: ['pkg', { }], arch: 'arch', signing: 'signing' }
```
Comment (erisu) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]:
```suggestion
                mac: { package: ['pkg', { dmg: { } }], arch: 'arch', signing: 'signing' }
```
The test case is checking the options OF the package, not the object that defines the package. This change is to clarify the test case.

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 2]
```diff
@@ -699,6 +699,152 @@ describe('Testing build.js:', () => {
...
+
+            electronBuilder = new ElectronBuilder(buildOptions, api).configureUserBuildSettings();
+
+            expect(existsSyncSpy).toHaveBeenCalled();
+            expect(requireSpy).toHaveBeenCalled();
+
+            const expectedMac = {
+                target: [
+                    { target: 'pkg', arch: 'arch' },
+                    { target: { }, arch: 'arch' }
```
Comment (erisu) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]:
```suggestion
                    { target: 'dmg', arch: 'arch' }
```
Related to earlier change.

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 3]
```diff
@@ -699,6 +699,152 @@ describe('Testing build.js:', () => {
...
+            expect(electronBuilder.userBuildSettings.config.mac).toEqual(expectedMac);
+            expect(electronBuilder.userBuildSettings.config.dmg).toEqual(undefined);
+            expect(electronBuilder.userBuildSettings.config.linux).toEqual(undefined);
+            expect(electronBuilder.userBuildSettings.config.windows).toEqual(undefined);
+        });
+
+        it('should append package top-level key options.', () => {
+            // mock platformConfig, buildConfig and buildOptions Objects
+            const platformConfig = {
+                mac: { package: ['pkg', { 'dmg': { 'format': 'UDZO' } }], arch: 'arch', signing: 'signing' }
```
Comment (erisu) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]:
```suggestion
                mac: { package: ['pkg', { dmg: { 'format': 'UDZO' } }], arch: 'arch', signing: 'signing' }
```

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 4]
```diff
@@ -699,6 +699,152 @@ describe('Testing build.js:', () => {
...
+            expect(electronBuilder.userBuildSettings.config.mac).toEqual(expectedMac);
+            expect(electronBuilder.userBuildSettings.config.dmg).toEqual(expectedDmgOptions);
+            expect(electronBuilder.userBuildSettings.config.linux).toEqual(undefined);
+            expect(electronBuilder.userBuildSettings.config.windows).toEqual(undefined);
+        });
+
+        it('should append package top-level key nested options.', () => {
+            // mock platformConfig, buildConfig and buildOptions Objects
+            const platformConfig = {
+                mac: { package: ['pkg', { 'dmg': { 'format': { 'UDZO': '' } } }], arch: 'arch', signing: 'signing' }
```
Comment (erisu) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]:
```suggestion
                mac: { package: ['pkg', { dmg: { 'format': { 'UDZO': '' } } }], arch: 'arch', signing: 'signing' }
```

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 5]
```diff
@@ -699,6 +699,152 @@ describe('Testing build.js:', () => {
...
+                target: [
+                    { target: 'pkg', arch: 'arch' },
+                    { target: 'dmg', arch: 'arch' }
+                ],
+                type: '${BUILD_TYPE}',
+                icon: '${APP_INSTALLER_ICON}'
+            };
+
+            const expectedDmgOptions = {
+                'format': { 'UDZO': '' }
```
Comment (erisu) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]:
```suggestion
                format: { UDZO: '' }
```

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 6]
```diff
@@ -699,6 +699,152 @@ describe('Testing build.js:', () => {
...
+                target: [
+                    { target: 'pkg', arch: 'arch' },
+                    { target: 'dmg', arch: 'arch' }
+                ],
+                type: '${BUILD_TYPE}',
+                icon: '${APP_INSTALLER_ICON}'
+            };
+
+            const expectedDmgOptions = {
+                'format': 'UDZO'
```
Comment (erisu) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]:
```suggestion
                format: 'UDZO'
```

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 7]
```diff
@@ -699,6 +699,152 @@ describe('Testing build.js:', () => {
...
+                target: [
+                    { target: 'pkg', arch: 'arch' },
+                    { target: 'dmg', arch: 'arch' }
+                ],
+                type: '${BUILD_TYPE}',
+                icon: '${APP_INSTALLER_ICON}'
+            };
+
+            expect(electronBuilder.userBuildSettings.config.mac).toEqual(expectedMac);
+            expect(electronBuilder.userBuildSettings.config.dmg).toEqual(undefined);
```
Comment (GedasGa) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]:
```suggestion
            expect(electronBuilder.userBuildSettings.config.dmg).toEqual({ });
```

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 8]
```diff
@@ -699,6 +699,152 @@ describe('Testing build.js:', () => {
             expect(electronBuilder.userBuildSettings.config.win).toEqual(expectedWin);
         });
 
+        it('should not append package top-level key options if object is empty.', () => {
```
Comment (GedasGa) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]:
```suggestion
        it('should append package top-level key options if the object is empty.', () => {
```

**Pull request merged**